### ISSUE TITLE
Definition list style/pattern

### ISSUE THREAD

**Issue by ncameronbritt** [patternfly/patternfly-next#1624]
Both OpenShift and Cloud Services have a need for something like a definition list (or "spec" or "set of facts") where there are a set of fields with their associated values. 

Looks like Kyle has done a version of this for Cloud services (see screenshot), but I wonder if this should be a component or pattern.  

Kyle's design
![Screen Shot 2019-03-25 at 11 57 15 AM](https://user-images.githubusercontent.com/8974593/54934606-2cf94b00-4ef5-11e9-8657-42a236b63a57.png)

Current OpenShift
![Screen Shot 2019-03-25 at 11 58 08 AM](https://user-images.githubusercontent.com/8974593/54934674-4a2e1980-4ef5-11e9-9783-e1892fd6e00d.png)

**Comment by rachael-phillips** [comment]
Thanks @ncameronbritt this would be a great topic to bring up in @mcarrano's Thursday design syncs where design contribution ideas are discussed. @ncameronbritt would you have any interest in presenting this at that meeting?

**Comment by matthewcarleton** [comment]
Has there been any discussion around edit mode for this view?

**Comment by mcarrano** [comment]
No, but I know this is a use case that often comes up. We should definitely consider this.

**Comment by ncameronbritt** [comment]
@matthewcarleton not exactly. Though I was discussing inline edit yesterday separately. There are issues open: #295 and #404.

Just off the top of my head, if there are only a small number of editable fields, then inline edit would work well. OTOH, if most/all fields needed to be editable then it seems like it would make sense to flip to a form. We already have patterns for forms and inline edit is in the works, so I would be disinclined to further complicate this pattern, but perhaps others have different views.

**Comment by mcarrano** [comment]
In the past, there have been questions about placement of buttons when I have a case where the form changes mode (read-only to edit and back). It would be good to address this as part of the guidance. Would be a good candidate for a demo IMO.

**Comment by matthewcarleton** [comment]
Ya I think there could be a good discussion/investigation into what's most logical here for editing. Happy to be more involved if that's helpful :)

**Comment by mcarrano** [comment]
@rachael-phillips @LHinson This is a candidate to place on the roadmap for further design work.

**Comment by rachael-phillips** [comment]
Hi @ncameronbritt what is the timeline and priorities for definition list?

**Comment by ncameronbritt** [comment]
I don't have a specific timeline for needing it. Up to now, I think that products that have needed this pattern have done their own thing, so I don't think anyone is waiting on it. But it does seem like something we could align on as products move to PF4.

**Comment by ncameronbritt** [comment]
I took a stab at the visual design for this. There are several iterations there, along with examples with OpenShift content. I'm happy to work with visual design on this more, or turn it over to them.
[](https://drive.google.com/file/d/1A4zYZNsOD2cB6_hho-LlGaUiHJGA9zIp/view?usp=sharing)

I started with the spacing/text styles of forms since this is sort of a read-only version of a form. One deviation is that I preferred the look of a lighter color for the labels. I'm not sure that there's a great reason to have two different treatments for labels. Happy for any feedback.

**Comment by rachael-phillips** [comment]
@mcarrano making you aware. Is this something we want to bring to the new Thursday sync format?

**Comment by mcarrano** [comment]
Thanks for working on this @ncameronbritt . So is it fair to say that this is basically just a multi-column read-only form?

@rachael-phillips I'd like to get Visuals to look at this before be put this up for a full review. We'll also need to consider how this responds as the viewport shrinks.

**Comment by ncameronbritt** [comment]
I'd say that's almost correct @mcarrano. Correct that it's sort of a read-only form, but it need not be multi-column. It just happens that some of my examples are multi-column. And like regular forms, there may be use cases for having labels top aligned or inline. 

I'm not sure there are any unique considerations for mobile--I would imagine the inline label version would behave just like inline forms.

**Comment by beanh66** [comment]
@mcarrano @rachael-phillips @LHinson 

Cameron shared the example of where this is an issue for us on all resource detail pages in OpenShift. Here is another look at the issue and how we could solve it with the lighter text for field labels (both for resource details and table columns). Below shows the field label issue we have on expanded list views: 

![Screen Shot 2019-05-02 at 3 20 07 PM](https://user-images.githubusercontent.com/18057044/57101192-aeab7800-6cee-11e9-9e79-dbda8d8c94ef.png)

**Comment by maryshak1996** [comment]
@beanh66 I think that the styling above works well to define hierarchy. I think that the use of a secondary color and a primary color for headers and copy, respectively, works well at providing subtle differentiators

**Comment by LHinson** [comment]
@kybaker plans to look at the text styling next week. 

@mcarrano also hoping we can move the needle on the data list pattern idea so we can get this into PF4. It still has the interaction tag, anything left there or should this move to visuals as you suggest above?

**Comment by mcarrano** [comment]
I'm finally having a chance to look at this and it seems like there are two separate issues. The primary request here is for a way to display read-only data in an easy to read presentation that makes it easy to differentiate between field labels and values. This would be useful for any use case that requires drill-down to details in an expansion panel, a page, or a modal. @LHinson agree that this can move to visuals to propose the appropriate styling and spacing. I will tag and reassign to @kybaker .

The issue of the table header styling is related, but would require a change to the Table component. @beanh66 are you suggesting that we change the styling of table headers, in general? It would be great if you can open a separate issue for this with any supporting data.

**Comment by beanh66** [comment]
@mcarrano yes that is the proposal. We are certainly more concerned about the former (differentiating between field labels and values) but we realized the same problem exists with the compact table since the headers and content use the same font size, color, and style.

**Comment by mcarrano** [comment]
@beanh66 I've opened a new issue here to address styling of the table column headings. Feel free to add further comments or background here.  @LHinson FYI

**Comment by maryshak1996** [comment]
reviewed this with @mcarrano , @kybaker , @mceledonia , and @ncameronbritt and we have established a direction for this.
The definition list should be created with the same spacing and flexibility to choose between inline and stacked layouts as a form does, but instead of a label and an input field, it would have a label and some sort of content (text, links, tags, etc.). 
The only update needed here is a more global change to label styling and column header styling in tables (as they all serve the same function of providing context for the part the user needs to actually interact with or read information from)
This proposed change is to style labels and column headers this way:
Red Hat font, 14pt, pf-global-Color-100, medium

My mockups for this pattern can be found here: 
https://redhat.invisionapp.com/share/VYS77JRM9EB

**Comment by beanh66** [comment]
LGTM @maryshak1996! Thanks for the update. @priley86 is currently making the update in product as he is working on the PF4 table updates now 🎉

**Comment by priley86** [comment]
thanks for reviewing this w/ me! Will be sure to share screenshots w/ @beanh66 and @maryshak1996 once ready for review.

**Comment by maryshak1996** [comment]
Here's my sketch file for this (the top section titled "FINAL DIRECTION" is what should be used) 
https://sketch.cloud/s/Yn1om?settings=Yn1om

**Comment by mcarrano** [comment]
Current designs are approved and ready for CSS implementation. @mcoker we are unsure whether this calls for a new component, i.e. a read-only attribute value pair or if this can be just constructed as a demo using existing components. Would like to get your input.

**Comment by mcarrano** [comment]
Met this morning with @mcoker and @maryshak1996 to talk about the approach here for implementation. We will tackle this in 2 parts:

* CReate a new Definition List component that is based on the Content component but allows a vertical as well as horizontal arrangement of the label-value pair. This will allow any element to be placed in the value area.

* CReate a demo that uses the new component to create definition list examples per @maryshak1996 's designs.

I have converted this to an epic and will create two new issues to track the above work.

**Comment by mcoker** [comment]
@mcarrano mind if we move this to patternfly-design? We've got https://github.com/patternfly/patternfly-next/issues/1919 to track the core work.

**Comment by mcarrano** [comment]
The design work is done @mcoker . I think this can just be closed.
